pkg/virt-controller/watch/export/export.go

```diff
@@ -0,0 +1,864 @@
+			vmExportCopy.Status.Phase = exportv1.Pending
+		}
+	}
+
+	if len(pvcs) == 0 {
+		log.Log.V(1).Info("PVC(s) not found, updating status to not found")
+		updateCondition(vmExportCopy.Status.Conditions, newPvcCondition(corev1.ConditionFalse, pvcNotFoundReason), true)
+	} else {
+		updateCondition(vmExportCopy.Status.Conditions, ctrl.pvcConditionFromPVC(pvcs), true)
+	}
+	internalCert, err := ctrl.base64EncodeExportCa()
+	if err != nil {
+		return retry, err
+	}
+	vmExportCopy.Status.ServiceName = service.Name
+	vmExportCopy.Status.Links = &exportv1.VirtualMachineExportLinks{}
+	vmExportCopy.Status.Links.Internal = &exportv1.VirtualMachineExportLink{
+		Volumes: []exportv1.VirtualMachineExportVolume{},
+		Cert:    internalCert,
+	}
+	for _, pvc := range pvcs {
+		if pvc != nil && exporterPod != nil && exporterPod.Status.Phase == corev1.PodRunning {
+			const scheme = "https://"
+			host := fmt.Sprintf("%s.%s.svc", service.Name, service.Namespace)
+			if ctrl.isKubevirtContentType(pvc) {
+				vmExportCopy.Status.Links.Internal.Volumes = append(vmExportCopy.Status.Links.Internal.Volumes, exportv1.VirtualMachineExportVolume{
+					Name: pvc.Name,
+					Formats: []exportv1.VirtualMachineExportVolumeFormat{
+						{
+							Format: exportv1.KubeVirtRaw,
+							Url:    scheme + path.Join(host, rawURI(pvc)),
+						},
+						{
+							Format: exportv1.KubeVirtGz,
+							Url:    scheme + path.Join(host, rawGzipURI(pvc)),
+						},
+					},
+				})
+			} else {
+				vmExportCopy.Status.Links.Internal.Volumes = append(vmExportCopy.Status.Links.Internal.Volumes, exportv1.VirtualMachineExportVolume{
+					Name: pvc.Name,
+					Formats: []exportv1.VirtualMachineExportVolumeFormat{
+						{
+							Format: exportv1.Dir,
+							Url:    scheme + path.Join(host, dirURI(pvc)),
+						},
+						{
+							Format: exportv1.ArchiveGz,
+							Url:    scheme + path.Join(host, archiveURI(pvc)),
+						},
+					},
+				})
+			}
+		}
+	}
+	if !equality.Semantic.DeepEqual(vmExport, vmExportCopy) {
+		if _, err := ctrl.Client.VirtualMachineExport(vmExportCopy.Namespace).Update(context.Background(), vmExportCopy, metav1.UpdateOptions{}); err != nil {
+			return retry, err
+		}
+	}
+	if vmExportCopy.Status.Phase == exportv1.Pending {
+		log.Log.V(4).Info("Not ready requeueing")
+		retry = time.Second
+	}
+	return retry, nil
+}
+
+func (ctrl *VMExportController) base64EncodeExportCa() (string, error) {
+	key := controller.NamespacedKey(ctrl.KubevirtNamespace, components.KubeVirtExportCASecretName)
+	ctrl.ConfigMapInformer.GetStore().GetByKey(key)
+	obj, exists, err := ctrl.ConfigMapInformer.GetStore().GetByKey(key)
+	if err != nil || !exists {
+		log.DefaultLogger().Infof("CA Config Map not found, %v", ctrl.ConfigMapInformer.GetStore().ListKeys())
+		return "", err
+	}
+	cm := obj.(*corev1.ConfigMap).DeepCopy()
+	bundle := cm.Data[caBundle]
+	return base64.StdEncoding.EncodeToString([]byte(bundle)), nil
```

Comment:
The proposal contains the following: `cert: <base64 encoded cert>` so that is why I base64 encoded it. If that is not something we want, I can not encode it.

Reply:
Okay, let me remove the base64 encoding, since like you said it is already pem encoded.